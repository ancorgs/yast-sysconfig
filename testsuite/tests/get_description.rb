# encoding: utf-8

# Sysconfig module - testsuite
#
# get_description() tests
#
# testedfiles: Sysconfig.ycp
#
# $Id$
#
module Yast
  class GetDescriptionClient < Client
    def main
      Yast.import "Testsuite"
      Yast.import "Pkg" # override packamanager
      Yast.import "Sysconfig"

      Testsuite.Test(lambda do
        Sysconfig.Set(
          [
            {
              "sysconfig_key"   => "VARIABLE",
              "sysconfig_path"  => "/etc/sysconfig/test",
              "sysconfig_value" => "no"
            }
          ]
        )
      end, [
        {},
        {},
        {}
      ], nil)


      # test non-existing variable
      Testsuite.Test(lambda { Sysconfig.get_description("nonexisting$variable") }, [
        {},
        {},
        {}
      ], nil)

      # test variable without metadata
      @READ = {
        "syseditor" => {
          "value"         => {
            "/etc/sysconfig/test" => { "VARIABLE" => "yes" }
          },
          "value_comment" => {
            "/etc/sysconfig/test" => {
              "VARIABLE" => "# Only comment, no metadata"
            }
          }
        }
      }
      Testsuite.Test(lambda do
        Sysconfig.get_description("VARIABLE$/etc/sysconfig/test")
      end, [
        @READ,
        {},
        {}
      ], nil)


      # test variable with metadata
      @READ = {
        "syseditor" => {
          "value"         => {
            "/etc/sysconfig/test" => { "VARIABLE" => "yes" }
          },
          "value_comment" => {
            "/etc/sysconfig/test" => {
              "VARIABLE" => "## Path: Test\n" +
                "## Description: help\n" +
                "## Type: yesno\n" +
                "# Comment with metadata"
            }
          }
        }
      }
      Testsuite.Test(lambda do
        Sysconfig.get_description("VARIABLE$/etc/sysconfig/test")
      end, [
        @READ,
        {},
        {}
      ], nil)


      # test multiple keywods in metadata
      @READ = {
        "syseditor" => {
          "value"         => {
            "/etc/sysconfig/test" => { "VARIABLE" => "yes" }
          },
          "value_comment" => {
            "/etc/sysconfig/test" => {
              "VARIABLE" => "## Path: Test\n" +
                "## Description: help\n" +
                "## Type: yesno\n" +
                "##Path: Second path definition\n" +
                "# Comment with metadata"
            }
          }
        }
      }
      Testsuite.Test(lambda do
        Sysconfig.get_description("VARIABLE$/etc/sysconfig/test")
      end, [
        @READ,
        {},
        {}
      ], nil)

      nil
    end
  end
end

Yast::GetDescriptionClient.new.main
