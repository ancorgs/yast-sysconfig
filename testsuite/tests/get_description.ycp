// Sysconfig module - testsuite
//
// get_description() tests
//
// testedfiles: Sysconfig.ycp
//
// $Id$
//

{

import "Testsuite";
import "Pkg";   // override packamanager
import "Sysconfig";

Testsuite::Test( ``(Sysconfig::Set([ $[ "sysconfig_key" : "VARIABLE",
    "sysconfig_path" : "/etc/sysconfig/test",
    "sysconfig_value" : "no" ] ]) ), [ $[], $[], $[] ], nil );


// test non-existing variable
Testsuite::Test( ``(Sysconfig::get_description("nonexisting$variable") ), [ $[], $[], $[] ], nil );

// test variable without metadata
map READ = $[ "syseditor" : $[ "value" : $[ "/etc/sysconfig/test" : $[ "VARIABLE" : "yes" ]],
			       "value_comment" : $[ "/etc/sysconfig/test" : $[ "VARIABLE" : "# Only comment, no metadata" ]]
			    ]
];
Testsuite::Test( ``(Sysconfig::get_description("VARIABLE$/etc/sysconfig/test") ), [ READ, $[], $[] ], nil );


// test variable with metadata
READ = $[ "syseditor" : $[ "value" : $[ "/etc/sysconfig/test" : $[ "VARIABLE" : "yes" ]],
			       "value_comment" : $[ "/etc/sysconfig/test" : $[ "VARIABLE" : "## Path: Test\n## Description: help\n## Type: yesno\n# Comment with metadata" ]]
			    ]
];
Testsuite::Test( ``(Sysconfig::get_description("VARIABLE$/etc/sysconfig/test") ), [ READ, $[], $[] ], nil );


// test multiple keywods in metadata
READ = $[ "syseditor" : $[ "value" : $[ "/etc/sysconfig/test" : $[ "VARIABLE" : "yes" ]],
			       "value_comment" : $[ "/etc/sysconfig/test" : $[ "VARIABLE" : "## Path: Test\n## Description: help\n## Type: yesno\n##Path: Second path definition\n# Comment with metadata" ]]
			    ]
];
Testsuite::Test( ``(Sysconfig::get_description("VARIABLE$/etc/sysconfig/test") ), [ READ, $[], $[] ], nil );

}

