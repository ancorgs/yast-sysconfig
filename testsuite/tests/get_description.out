Return	nil
Read	.syseditor.value_comment."variable"."nonexisting" nil
Read	.syseditor.value."variable"."nonexisting" nil
Return	$["comment":"", "file":"variable", "location":"", "name":"nonexisting", "value":nil]
Read	.syseditor.value_comment."/etc/sysconfig/test"."VARIABLE" "# Only comment, no metadata"
Read	.syseditor.value."/etc/sysconfig/test"."VARIABLE" "yes"
Return	$["comment":"Only comment, no metadata ", "file":"/etc/sysconfig/test", "location":"", "name":"VARIABLE", "new_value":"no", "value":"yes"]
Read	.syseditor.value_comment."/etc/sysconfig/test"."VARIABLE" "## Path: Test\n## Description: help\n## Type: yesno\n# Comment with metadata"
Read	.syseditor.value."/etc/sysconfig/test"."VARIABLE" "yes"
Return	$["Description":"help", "Path":"Test", "Type":"yesno", "comment":"Comment with metadata ", "file":"/etc/sysconfig/test", "location":"", "name":"VARIABLE", "new_value":"no", "value":"yes"]
Read	.syseditor.value_comment."/etc/sysconfig/test"."VARIABLE" "## Path: Test\n## Description: help\n## Type: yesno\n##Path: Second path definition\n# Comment with metadata"
Read	.syseditor.value."/etc/sysconfig/test"."VARIABLE" "yes"
Return	$["Description":"help", "Path":"Second path definition", "Type":"yesno", "comment":"Comment with metadata ", "file":"/etc/sysconfig/test", "location":"", "name":"VARIABLE", "new_value":"no", "value":"yes"]
